includes/admin/reports/class-wc-report-downloads.php

```diff
@@ -0,0 +1,256 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+if ( ! class_exists( 'WP_List_Table' ) ) {
+	require_once( ABSPATH . 'wp-admin/includes/class-wp-list-table.php' );
+}
+
+/**
+ * WC_Report_Downloads.
+ *
+ * @author      WooThemes
+ * @category    Admin
+ * @package     WooCommerce/Admin/Reports
+ * @version     3.3.0
+ */
+class WC_Report_Downloads extends WP_List_Table {
+
+	/**
+	 * Max items.
+	 *
+	 * @var int
+	 */
+	protected $max_items;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+
+		parent::__construct( array(
+			'singular'  => 'download',
+			'plural'    => 'downloads',
+			'ajax'      => false,
+		) );
+	}
+
+	/**
+	 * Don't need this.
+	 *
+	 * @param string $position
+	 */
+	public function display_tablenav( $position ) {
+
+		if ( 'top' !== $position ) {
+			parent::display_tablenav( $position );
+		}
+	}
+
+	/**
+	 * Output the report.
+	 */
+	public function output_report() {
+
+		$this->prepare_items();
+
+		// Subtitle for permission if set.
+		if ( isset( $_GET['permission_id'] ) && is_numeric( $_GET['permission_id'] ) && $_GET['permission_id'] > 0 ) {
+			$permission_id = $_GET['permission_id'];
+			// Load the permission, order, etc. so we can render more information.
+			$permission	= null;
+			$product	= null;
+			try {
+				$permission	= new WC_Customer_Download( $permission_id );
+				$product	= wc_get_product( $permission->product_id );
+			} catch ( Exception $e ) {
```

Comment:
We're using a generic exception here; will that always mean that the permission wasn't found? Or could other errors creep in here too? (Note: I haven't looked too closely into either `WC_Customer_Download` or `wc_get_product`)